src/components/code-diff-editor/code-diff-editor.tsx

```diff
@@ -0,0 +1,223 @@
+  @Watch("yamlSchemaUri")
+  yamlSchemaUriChange(newUri: string) {
+    if (this.language !== "yaml") {
+      return;
+    }
+
+    // Necessary to not combine the current scheme with the new scheme
+    this.#editorId = `ch-diff-editor-${autoId++}`;
+
+    if (newUri) {
+      configureMonacoYaml(monaco, {
+        enableSchemaRequest: true,
+        format: true,
+        schemas: this.#getYamlSchemas()
+      });
+    } else {
+      configureMonacoYaml(monaco, {
+        enableSchemaRequest: true,
+        format: true,
+        schemas: []
+      });
+    }
+
+    this.#monacoDiffEditorInstance.setModel({
+      original: monaco.editor.createModel(
+        this.value,
+        this.language,
+        monaco.Uri.parse(`file:///${this.#editorId}.txt`)
+      ),
+      modified: monaco.editor.createModel(
+        this.modifiedValue,
+        this.language,
+        monaco.Uri.parse(`file:///${this.#editorId}-modified.txt`)
+      )
+    });
+  }
+
+  #getYamlSchemas = () => [
+    {
+      // If YAML file is opened matching this glob
+      fileMatch: [`**/${this.#editorId}.*`],
+      // Then this schema will be downloaded from the internet and used.
+      uri: this.yamlSchemaUri
+    },
+    {
+      // If YAML file is opened matching this glob
+      fileMatch: [`**/${this.#editorId}-modified.*`],
+      // Then this schema will be downloaded from the internet and used.
+      uri: this.yamlSchemaUri
+    }
+  ];
+
+  #setupDiffEditor() {
+    this.#monacoDiffEditorInstance = monaco.editor.createDiffEditor(
+      this.#monacoRef,
+      {
+        ...this.options,
+        ...(!(
+          "originalEditable" in this.options || "readOnly" in this.options
+        ) && {
+          originalEditable: !this.readonly,
+          readOnly: this.readonly
+        })
+      }
+    );
+
+    this.#monacoDiffEditorInstance.setModel({
+      original: monaco.editor.createModel(
+        this.value,
+        this.language,
+        monaco.Uri.parse(`file:///${this.#editorId}.txt`)
+      ),
+      modified: monaco.editor.createModel(
+        this.modifiedValue,
+        this.language,
+        monaco.Uri.parse(`file:///${this.#editorId}-modified.txt`)
+      )
+    });
+  }
+
+  #configureYaml() {
+    if (this.language === "yaml" && !!this.yamlSchemaUri) {
+      configureMonacoYaml(monaco, {
+        enableSchemaRequest: true,
+        format: true,
+        schemas: this.#getYamlSchemas()
+      });
+    }
+  }
+
+  connectedCallback(): void {
+    this.#editorId = `ch-diff-editor-${autoId++}`;
+  }
+
+  componentDidLoad() {
+    this.#configureYaml();
+    this.#setupDiffEditor();
+
+    this.#resizeObserver = new ResizeObserver(entries => {
+      const absoluteContentEntry = entries[0].contentRect;
+
+      this.#monacoDiffEditorInstance.layout({
+        width: absoluteContentEntry.width,
+        height: absoluteContentEntry.height
+      });
+    });
+
+    this.#resizeObserver.observe(this.#absoluteContentRef);
+  }
```

Comment:
I notice that in the previous implementation we didn't disconnect the observer to avoid memory leaks. Can you add this change in the `disconnectedCallback` method?

For reference: https://github.com/genexuslabs/chameleon-controls-library/blob/main/src/components/popover/popover.tsx#L1067